### PR TITLE
sphinxsearch: Add support for MySQL & xmlpipe2

### LINKED ISSUE #70076: sphinxsearch needs expat for xmlpipe2 support
**Describe the bug**
To enable xmlpipe2 support `expat` needs to be in nativeBuildInputs for `sphinxsearch` expression.

**To Reproduce**
1. Run `sphinxsearch-indexer` with xmlpipe2 source in configuration file.
2. Observe an error that XML support is missing.

**Expected behavior**
xmlpipe2 sources should work out of the box.

**Metadata**
```
 - system: `"i686-linux"`
 - host os: `Linux 4.19.72, NixOS, 19.03.173489.af5481e9b77 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by vvs-**
###### Motivation for this change

Sphinxsearch doesn't support MySQL sources and xmlpipe2
Closes #70076

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/search/sphinxsearch/default.nix [hunk 1]
```diff
@@ -13,12 +13,17 @@ stdenv.mkDerivation {
...
     "--program-prefix=sphinxsearch-"
-    "--without-mysql"
+    "--with-mysql=${mariadb}/"
+    "--with-libexpat"
     "--enable-id64"
   ];
 
   nativeBuildInputs = [
-    pkgconfig
+    pkgconfig expat
```
Comment (veprbl) [inline pkgs/servers/search/sphinxsearch/default.nix]: If you link with expat, then it should go to `buildInputs`?

Comment (vvs-) [reply pkgs/servers/search/sphinxsearch/default.nix]: I'm not an expert, so it's up to you.

Comment (vvs-) [reply pkgs/servers/search/sphinxsearch/default.nix]: Fixed.

File: pkgs/servers/search/sphinxsearch/default.nix [hunk 2]
```diff
@@ -13,14 +13,19 @@ stdenv.mkDerivation {
...
+    "--with-libexpat"
     "--enable-id64"
   ];
 
   nativeBuildInputs = [
     pkgconfig
   ];
 
+  buildInputs = [
+    expat mariadb zlib openssl
```
Comment (jtojnar) [inline pkgs/servers/search/sphinxsearch/default.nix]: Could you make this one per line? And mention why is zlib or openssl needed?

**Review by jtojnar — CHANGES_REQUESTED** [review]
Could you follow out [commit style](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)? For example, there could be two commits named like:
- sphinxsearch: Add support for xmlpipe2
- sphinxsearch: Add support for MySQL sources

**Comment by jtojnar** [comment]
Also could you evaluate the changes in closure size as suggested in the OP:

>  Determined the impact on package closure size (by running `nix path-info -h -S $(nix-build --no-out-link -A sphinxsearch)` before and after)

File: pkgs/servers/search/sphinxsearch/default.nix (hunk 2 above)
Comment (vvs-) [reply pkgs/servers/search/sphinxsearch/default.nix]: I should reiterate that I'm not an expert. I don't know exactly why those libraries are needed, but without them configure test for MySQL presence will fail as they are linked there after mariadb.

**Comment by vvs-** [comment]
I'm not sure if I will have time as this is not a priority for me.

BTW don't take it personally, but shouldn't package maintainers care for such things? And I already gave anyone on nixpkgs team write access to my repository branch if they feel inclined to work on it.

**Comment by Valodim** [comment]
I'll pick this up and make a PR in the next few days.

**Comment by vvs-** [comment]
You are welcome. I'm willing to help, but I can't afford to do everything alone.

**Pull request closed without merging**